pkg/metrics/builder/builder.go

```diff
@@ -0,0 +1,130 @@
+package builder
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// MetricBuilder is the basic metric type that must be used by any potential metric added to
+// the system. It allows user to create a prom collector which will be used to push.
+type MetricBuilder struct {
+	opts MetricOpts
+}
```

Comment:
tying the opts for metric builder to the definition of metric builder is not best imo. the metric opts need to be designed for creating a builder easily, while the builder is technically supposed to do more.

Reply:
I'm not sure I understand the issue with embedding `opts`. The intention was to have a single definition of the fields, rather than needing to handle them in the MetricsBuilder struct, initializer, and--now--opts struct. 

That said, this is not a sticking point, I think it is manageable as is, but still think it would be neater by embedding `opts`.